src/rdo.rs

```diff
@@ -214,15 +214,14 @@ pub fn rdo_mode_decision(
   for &luma_mode in &mode_set {
     assert!(fi.frame_type == FrameType::INTER || luma_mode.is_intra());
 
-    let same_as_luma_modes = &[ luma_mode ];
-    let mode_set_chroma = if is_chroma_block && luma_mode.is_intra() {
-      intra_mode_set
-    } else {
-      same_as_luma_modes
-    };
+    let mut mode_set_chroma = vec![ luma_mode ];
+
+    if is_chroma_block && luma_mode.is_intra() && luma_mode != PredictionMode::DC_PRED {
+      mode_set_chroma.push(PredictionMode::DC_PRED);
+    }
 
     // Find the best chroma prediction mode for the current luma prediction mode
-    for &chroma_mode in mode_set_chroma {
+    for &chroma_mode in &mode_set_chroma {
```

Comment:
For given luma_mode, it now checks only one chroma_mode, so it does not need for loop here?

Reply:
Sometimes there should be a second mode (see push() on line 220)

Reply:
Oh right- you mentioned it in the commit message as well. If luma is not DC then chroma rdo loop test DC and the non-DC mode.